.github/workflows/tests_without_binary.yml

```diff
@@ -34,19 +34,19 @@ jobs:
       - name: Get required Python packages
         run: |
           cd main
-          python -m pip install --upgrade pip
+          python -m pip install pip==23.0
```

Comment:
The most recent pip version has some problems with editable pip installation.
This was breaking this CI.

Reply:
Do we need this version pinned in req.txt too?

Reply:
Yes. I pinned pip for now.